File: scala/gigasecond/src/main/scala/gigasecond.scala
import java.util.{Calendar}

class Gigasecond(birthday: Calendar) {
  import Gigasecond.GigasecondCalendar

  def date(): Calendar = {
    birthday addSeconds Gigasecond.Duration
  }
}

object Gigasecond {

  val Duration = BigInt(10).pow(9).toInt

  def apply(birthday: Calendar) = new Gigasecond(birthday)

  implicit class GigasecondCalendar(date: Calendar) {

    def addSeconds(duration: Int): Calendar = {
      val result = date.clone().asInstanceOf[Calendar]
      result.add(Calendar.SECOND, duration)
      result
    }

  }
}

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {

  private static long GIGASECOND = 1000000000;
  private LocalDateTime date;

  public Gigasecond(LocalDate origin) {
    this(origin.atStartOfDay());
  }

  public Gigasecond(LocalDateTime origin) {
    this.date = origin.plusSeconds(GIGASECOND);
  }

  public LocalDateTime getDate() {
    return this.date;
  }

}